### PR TITLE
Transform resize

### LINKED ISSUE #90: Transform entity when selected (resize)

### PR DESCRIPTION
**Pull request by tlein**
Resolves #90 

Engine breaking changes:
* ShapeDrawable circles are now ovals that have a radiusWidth and radiusHeight instead of just radius
* Collision attributes now have a scale property
* Rotation transforms are always applied before scaling transforms

This will also break the visual aspect of the cross in the template game

### CODE REVIEW

File: src/duckling/canvas/tools/entity-move-tool.ts [hunk 1]
```diff
@@ -21,30 +23,46 @@ import {BaseTool, CanvasMouseEvent, CanvasKeyEvent} from './base-tool';
 @Injectable()
 export class EntityMoveTool extends BaseTool {
     private _selection : EntityKey;
+    private _selectedCorner : Vector = null;
```
Comment (jeffswenson) [inline src/duckling/canvas/tools/entity-move-tool.ts:25]: Why are you initializing these to null?

File: src/duckling/entitysystem/services/entity-eligible-resize.service.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import {Injectable} from '@angular/core';
+
+import {BaseAttributeService} from '../base-attribute.service';
+import {AttributeKey, Entity} from '../entity';
+
+export type AttributeEligibleForResize = () => boolean;
+
+@Injectable()
+export class EntityEligibleResizeService extends BaseAttributeService<AttributeEligibleForResize> {
```
Comment (jeffswenson) [inline src/duckling/entitysystem/services/entity-eligible-resize.service.ts]: I think this should be a function on the Resize service. It should return true if the entity has any implementations for resize.

File: src/duckling/game/collision/collision-eligible-for-resize.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export function collisionIsEligibleForResize() {
```
Comment (jeffswenson) [inline src/duckling/game/collision/collision-eligible-for-resize.ts]: You can delete this if you listen to my other comment.

File: src/duckling/canvas/tools/entity-move-tool.ts [hunk 4]
```diff
@@ -75,43 +95,137 @@ export class EntityMoveTool extends BaseTool {
         this._cancel();
     }
 
-    get key() {
-        return "EntityMoveTool";
+    private _processCornerResize(currentCoords : Vector) {
+        let selectedEntityKey = this._selectionService.selection.value.selectedEntity;
```
Comment (jeffswenson) [inline src/duckling/canvas/tools/entity-move-tool.ts:101]: You should extract as much code as you can into stateless functions in another file. Consider naming it something like selection-box.ts

File: src/duckling/game/collision/collision-size.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import {Vector, vectorAbsolute, vectorSign, vectorMultiply, vectorDivide} from '../../math/vector';
+import {Entity} from '../../entitysystem/entity';
+import {immutableAssign} from '../../util/model';
+import {AssetService} from '../../project/asset.service';
+
+import {CollisionAttribute, getCollision} from './collision-attribute';
+import {collisionBoundingBox} from './collision-box';
+
+export function setCollisionSize(entity : Entity, newSize : Vector, assetService : AssetService) : CollisionAttribute {
```
Comment (jeffswenson) [inline src/duckling/game/collision/collision-size.ts]: This and the other set functions are good candidates for unit testing. They are complex and pure.

**Review by jeffswenson — CHANGES_REQUESTED** [review]

File: src/duckling/entitysystem/services/entity-eligible-resize.service.ts (hunk 2 above)
Comment (tlein) [reply src/duckling/entitysystem/services/entity-eligible-resize.service.ts]: fixed

File: src/duckling/game/collision/collision-eligible-for-resize.ts (hunk 3 above)
Comment (tlein) [reply src/duckling/game/collision/collision-eligible-for-resize.ts]: fixed

**Pull request closed without merging**